app/services/payout_report_publisher_includer.rb

```diff
@@ -60,6 +61,30 @@ def perform
       end
     end
 
+    puts @publisher.partner?
+    if @publisher.partner?
+      unless should_only_notify?
+        puts 'helo'
+        invoice = Invoice.find_by_partner_id(@publisher.id)
+        amount = invoice.finalized_amount_to_probi
+        PotentialPayment.create(payout_report_id: @payout_report.id,
+                                name: @publisher.name,
+                                amount: amount,
+                                fees: "0",
+                                publisher_id: @publisher.id,
+                                kind: PotentialPayment::MANUAL,
+                                address: "#{wallet.address}",
+                                uphold_status: uphold_status,
+                                reauthorization_needed: reauthorization_needed,
+                                uphold_member: uphold_member,
+                                suspended: suspended,
+                                uphold_id: uphold_id,
+                                invoice_id: invoice.id,
+                                finalized_by_id: invoice.finalized_by_id
+                                )
+      end
+    end
+
```

Comment:
If a publisher was also a partner, I think this may attempt to create two potential referral payments for the same balance.  I think it would fail since we have restrictions on the number of referral payments a single publisher can have for a given payout report.  I think what we want is to conditionally include these extra fields (invoice_id, finalized_by_id, etc) to the referral potential payment depending on if the publisher is a partner or not.  


https://github.com/brave-intl/publishers/blob/98fb9788542ebc16fcd746f9836f82c2b7b73975/app/models/potential_payment.rb#L25

Reply:
Thanks very much for the feedback @nvonpentz . Given our talk with @mandar-brave it seems there will be two payment flows, the regular one we are used to, and this manual one which addresses infinite codes. 

I took your advice and added a new `manual` parameter to `EnqueuePublishersForPayoutJob` which decides which type of potential payments to create. 

One thing we should think about is when codes become polymorphic, we should make sure they are not double counted in regular and manual payments. But for now I think we are okay. 

By the way, you had a good point that it would be tedious to generate a report for each partner individually. With the new `manual` parameter you can generate a report for all partners who currently have a `finalized invoice` 👍